scripts/globals/pets/avatar.lua

```diff
@@ -0,0 +1,501 @@
+local setMagicCastCooldown = function(pet)
+    local castingCooldown = 45
+    local master = pet:getMaster()
+
+    -- we already know master is a PC, this is just in case the pet gets disjointed somehow
+    if not master then
+        return
+    end
+
+    castingCooldown = castingCooldown - math.floor(xi.summon.getSummoningSkillOverCap(pet) / 3)
+
+    if master:hasStatusEffect(xi.effect.ASTRAL_FLOW) then
+        castingCooldown = castingCooldown - 5
+    end
+
+    local masterLegs = master:getEquipID(xi.slot.LEGS)
+    if
+        masterLegs and
+        (masterLegs == xi.item.SUMMONERS_SPATS or
+        masterLegs == xi.item.SUMMONERS_SPATS_P1 or
+        masterLegs == xi.item.SUMMONERS_SPATS_P2)
+    then
+        castingCooldown = castingCooldown - 5
+    end
+
+    local petElement = master:getPetElement()
+
+    -- AFAIK unaffected by scholar storms on the player
+    local actorWeather = pet:getWeather()
+    -- Strong weathers.
+    if
+        actorWeather == xi.combat.element.strongSingleWeather[petElement] or
+        actorWeather == xi.combat.element.strongDoubleWeather[petElement]
+    then
+        castingCooldown = castingCooldown - 2
+    -- Weak weathers.
+    elseif
+        actorWeather == xi.combat.element.weakSingleWeather[petElement] or
+        actorWeather == xi.combat.element.weakDoubleWeather[petElement]
+    then
+        castingCooldown = castingCooldown + 2
+    end
+
+    local dayElement = VanadielDayElement()
+    -- Strong day.
+    if dayElement == petElement then
+        castingCooldown = castingCooldown - 3
+    -- Weak day.
+    elseif dayElement == xi.combat.element.weakDay[petElement] then
+        castingCooldown = castingCooldown + 3
+    end
+
+    -- "Buff mode" is enabled for first cast after spawn and after casting any enhancing magic
+    if pet:getLocalVar(buffModeVar) ~= 1 then
+        castingCooldown = math.floor(castingCooldown / 2)
+    end
+
+    -- cast delay is ~1s past the finish of last spell, so we add casting time (or time since spell interrupt) to the castingCooldown
+    -- this is done by simply tracking the elapsed time since action is no longer xi.action.MAGIC_CASTING
+    local lastCastTime = pet:getLocalVar(lastCastTimeVar)
+    local lastCastTimeStamp = pet:getLocalVar(lastCastTimeStampVar)
+    if
+        lastCastTimeStamp > 0 and
+        os.time() - lastCastTimeStamp > lastCastTime
+    then
+        lastCastTime = os.time() - lastCastTimeStamp
+    end
+
+    if
+        lastCastTime > 0 and
+        pet:getCurrentAction() ~= xi.action.MAGIC_CASTING
+    then
+        pet:setLocalVar(lastCastTimeStampVar, 0)
+        pet:setLocalVar(lastCastTimeVar, lastCastTime)
+    end
+
+    pet:setMobMod(xi.mobMod.MAGIC_COOL, lastCastTime + castingCooldown)
+end
+
+xi.pets.avatar.onMobSpawn = function(pet)
+    local master = pet:getMaster()
+    if
+        not master or
+        master:getObjType() ~= xi.objType.PC
```

Comment:
Are there mobs that can cast light spirit and have it do things?

Reply:
Yes there are, I am not sure how mob light spirits are supposed to behave

Reply:
a typical yagudo's elemental (just for example - each mob race has a few different elements they can use, I forget who has light besides tonberry) will prioritize nuking you regardless of element - I'm not sure if they can even buff, they don't normally live long enough to find out. the ones summoned by the one tonberry NM for carby mitts have more complicated actions akin to a players light spirit (I've seen it cure). if anyone wants additional info, I can go hold some elementals to find out just let me know.

Reply:
fwiw it sounds like in both of those cases it behaves just like they currently do without my updated logic. Just randomly choosing spells from their spell list